### PR TITLE
BR-2347-update-refund-statuses

### PR DESCRIPTION
**Pull request by dhiraj-toptal**

### CODE REVIEW

**Comment by raimonbal** [comment]
please update the PR so doesn't have conflicts

**Comment by dhiraj-toptal** [comment]
PR updated with conflict fixes, can you review?

File: Controller/Webhook/Transaction/RefundCanceled.php [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+    /**
+     * CreateCsrfValidationException
+     *
+     * @param RequestInterface $request
+     * @return InvalidRequestException|null
+     */
+    public function createCsrfValidationException(RequestInterface $request): ?InvalidRequestException
+    {
+        return null;
```
Comment (raimonbal) [inline Controller/Webhook/Transaction/RefundCanceled.php:134]: null?

File: Controller/Webhook/Transaction/RefundFailed.php [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    /**
+     * CreateCsrfValidationException
+     *
+     * @param RequestInterface $request
+     * @return InvalidRequestException|null
+     */
+    public function createCsrfValidationException(RequestInterface $request): ?InvalidRequestException
+    {
+        return null;
```
Comment (raimonbal) [inline Controller/Webhook/Transaction/RefundFailed.php:134]: same as before

**Review by raimonbal — APPROVED** [review]
lgtm

File: Controller/Webhook/Transaction/RefundFailed.php (hunk 2 above)
Comment (dhiraj-toptal) [reply Controller/Webhook/Transaction/RefundFailed.php:134]: This starndard function and return value used for these CSRF process by magento.

**Pull request merged**